### ISSUE TITLE
Now they're doing email verification

### ISSUE THREAD

**Issue by KathrynElrod** [SeanDaBlack/KelloggBot#50]
This seems much harder to get around, yeah?

**Comment by bolshoytoster** [comment]
@KathrynElrod i'm surprised they didn't do this from the start, I think your suggestion of using guerrillamail will be fine until they start filtering out domains.

I wouldn't be surprised if there was a Kellogs dev going through these repos.

**Comment by KathrynElrod** [comment]
at least we're costing them headache and money, haha

**Comment by cafeclimber** [comment]
Just from quick searching, it looks like [10minutemail](https://10minutemail.com/) give a random domain? Idk how easy integration would be though

**Comment by bolshoytoster** [comment]
@cafeclimber we could use a combination of multiple of them.

Currently this issue is top priority so I'm starting work on a fix now.

**Comment by johnmarston139** [comment]
branch? @bolshoytoster

**Comment by bolshoytoster** [comment]
@johnmarston139 

> branch? @bolshoytoster 

I'm currently working on [my own fork](https://github.com/bolshoytoster/KelloggBot) but I could move it to a branch if you'd prefer.

**Comment by KathrynElrod** [comment]
https://10minutemail.net/address.api.php

currently gets a brand new address each time it's called, but there must be a way to pass in a user as a parameter.

**Comment by johnmarston139** [comment]
no worries, was just curious how you were bypassing the email verification. since this project is Python based and we all love 10minutemail, this may be useful https://pypi.org/project/python-10minutemail/

Scrape the inbox for the one-time code and send it to verify email; continue execution :) 

```
from minutemail import Mail
import time

# Create a new 10 minute mail
mail = Mail()
print(mail)

# Keep on checking for a new mail/message
while True:
    if mail.new_message():  # Check for new mail
        print(mail.fetch_message())  # Fetch all the messages
    time.sleep(2)
```

**Comment by pws1453** [comment]
> @johnmarston139
> 
> > branch? @bolshoytoster
> 
> I'm currently working on [my own fork](https://github.com/bolshoytoster/KelloggBot) but I could move it to a branch if you'd prefer.

I know some work has been done on my fork - I don't mind making a few diff branches if needed.

**Comment by pws1453** [comment]
This branch exists if you need it - should be able to accept PRs for a while
https://github.com/pws1453/KelloggBot/tree/email-verification

**Comment by KathrynElrod** [comment]
> https://pypi.org/project/python-10minutemail/

@johnmarston139 That looks basically perfect, nice find!

**Comment by tabrown2000** [comment]
FYI: the bottom of the python-10minutemail github has this warning:

> Warning: If you overuse this tool/API, then 10minutemail.com may block your IP address.

Just switching VPN servers might not be sufficient. Maybe the proxy server thing that some people have been experimenting with? Idk if we wanna abuse 10minutemail though, they're not the capitalist scum that we're after

**Comment by bolshoytoster** [comment]
@tabrown2000 I'm still working on the guerrilla mail verification as well, that doesn't say anything about blocking, the closest I could find was:

> Note: The API may be rate limited, but we do not publish our rate limits. Please use this API carefully.

If we alternate between the two providers it should ease the strain on their servers.

**Comment by johnmarston139** [comment]
> @tabrown2000 I'm still working on the guerrilla mail verification as well, that doesn't say anything about blocking, the closest I could find was:
> 
> > Note: The API may be rate limited, but we do not publish our rate limits. Please use this API carefully.
> 
> If we alternate between the two providers it should ease the strain on their servers.

I like this solution! Alternating between 2 providers may spread the load enough

**Comment by KathrynElrod** [comment]
We could also way lower the rate that the tool runs at. With even just 20 people running the script every 10 minutes, that's 120 applications an hour, which would still be enough to mess up their system.

**Comment by johnmarston139** [comment]
> 

I havent actually gotten a chance to play around w/ the package yet, just found it doing a quick preliminary search. Apologies if it's borked :P I'm all for w/e the better solution is

**Comment by tabrown2000** [comment]
Here's a stretch - I found a github repo where someone used Go to create their own guerilla mail server. Just throwin' that out there, I don't even know Go or have a computer to dedicate as a server, but...

https://github.com/flashmob/go-guerrilla

**Comment by bolshoytoster** [comment]
@tabrown2000 
> At a glance, guerilla mail looks like the best way to go. Not only does the python package have a built in method for switching servers, but also because I can't seem to get the 10minutemail python package to work. 
> 
> 
> 
> Anyone else run into a JSONDecodeError just trying to initialize the Mail class in the 10minutemail package?

I didn't know there's a python module for guerrilla, I was just sending the api requests manually.

I've nearly finished doing it this way so I'll submit a pr when I do.

**Comment by bolshoytoster** [comment]
@pws1453 I've opened the [pr](https://github.com/pws1453/KelloggBot/pull/3) if you want to test it out.

**Comment by pws1453** [comment]
it's pulled - was in a meeting

**Comment by bolshoytoster** [comment]
@pws1453 no worries.

**Comment by cafeclimber** [comment]
Hopefully I'm being helpful and not just spamming a ton of different services, but I also found [mail.tm](https://docs.mail.tm/) which has an api for numerous languages (including Python). From what I can tell the only restriction is 8 queries per second per IP which seems more than reasonable. I'm not sure the domains are entirely random, but the api is [here](https://github.com/CarloDePieri/pymailtm) (last commit 9 mos ago). I'll try to do some testing tonight and give updates on the various apis I've mentioned.

**Comment by bolshoytoster** [comment]
@cafeclimber 
> Hopefully I'm being helpful and not just spamming a ton of different services

Don't worry, any suggestion is useful.

I'll see if I can add 10minutemail support now and I'll try mail.tm later if I have time.

**Comment by cafeclimber** [comment]
@bolshoytoster I managed to get it working with mail.tm really easily. The only issue is it takes about 3-4 minutes for mail to come in.

Here's a minimal working example:

```python
import pymailtm

mail = pymailtm.MailTm()
acct = mail.get_account() # This randomly generates an account and password

while len(acct.get_messages()) == 0:
    sleep(1)

msgs = acct.get_messages()

print(msgs[0].text)
```

**Comment by bolshoytoster** [comment]
@cafeclimber I didn't see the `get_account` function sorry. It takes ~10 seconds with guerrilla.

If you think the 3-4 minutes is worth it I can add this to [the email-verification branch](https://github.com/pws1453/KelloggBot/tree/email-verification)

**Comment by cafeclimber** [comment]
No worries! Just wanted to make sure all the options were available. I think it could just be a backup option in the event domains start getting blocked, excessive guerilla queries, etc.

**Comment by bolshoytoster** [comment]
@cafeclimber I can implement this as a `try: except` where it could fall back to mail.tm if there's an error.

**Comment by KathrynElrod** [comment]
@pws1453 @bolshoytoster Awesome work, y’all - once you’re happy with it feel free to open a PR and I can review & merge

**Comment by bolshoytoster** [comment]
@KathrynElrod I've literally just finished adding mail.tm support.

My fork has some experimental stuff from the #25 fork, so should I open the pr to here or to that fork?

**Comment by pws1453** [comment]
I can merge you into mine now, and we can merge mine if ya want

**Comment by bolshoytoster** [comment]
@pws1453 alright, I'll open a pr to your `email-support` branch then you can merge that to main and hopefully merge the rest with this repo.

**Comment by pws1453** [comment]
okay cool send it over

**Comment by bolshoytoster** [comment]
@pws1453 I've opened it https://github.com/pws1453/KelloggBot/pull/4

**Comment by pws1453** [comment]
merged into my main, status reflected in https://github.com/SeanDaBlack/KelloggBot/pull/25

**Comment by bolshoytoster** [comment]
@pws1453 cheers, I'm going to go to bed now as it's 3:35 am.

**Comment by KathrynElrod** [comment]
Closed by #25
